File: TwainSaneInterface/src/DSEntry.swift
import Twain
import DataSource

var datasource: DataSource

func DS_Entry(
    pOrigin: Twain.Identity,
    DG: Int,
    DAT: Int,
    message: Int,
    pData: Twain.MemoryReference) -> Int {

    return datasource.Entry(pOrigin, DG, DAT, message, pData)
}

File: TwainSaneInterface/src/DSEntry.cpp
#include <TWAIN/TWAIN.h>
#include "DataSource.h"

DataSource datasource;

TW_UINT16 DS_Entry (pTW_IDENTITY pOrigin,
                    TW_UINT32    DG,
                    TW_UINT16    DAT,
                    TW_UINT16    MSG,
                    TW_MEMREF    pData) {

    return datasource.Entry (pOrigin, DG, DAT, MSG, pData);
}
